core/src/com/ericc/the/game/map/Map.java

```diff
@@ -108,18 +108,31 @@ public int height() {
      * It does REMOVE the passable tile it is going to return from the passableTiles collection!
      */
     public GridPoint2 getRandomPassableTile() {
-        GridPoint2 ret = passableTiles.iterator().next();
+        GridPoint2 ret;
+
+        try {
+            ret = passableTiles.iterator().next();
+        } catch (Exception e) {
+            throw new IllegalStateException("Can't find room for more entities, check the map size.");
```

Comment:
dodałbym tutaj TODO, bo in the long run to nie jest zachowanie którego byśmy chcieli. Najlepiej by było zwracać jakieś Optional<GridPoint2> i niech wołający sam sobie sprawdzi czy to miejsce jest czy nie ma. Jak rzucimy w niego exceptionem to już prawie na pewno kaplica, a to jest exception który jednak CHCIELIBYŚMY obsłużyć w runtimie.